ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -244,4 +261,33 @@ public File getRoot() {
       return safeSpace.directory;
     }
   }
+
+  private void verifyCleanShutdown(final File rootDirectory) {
```

Comment:
This logic appears to do more than just verify, it also deletes if the disk is believed to be unclean.